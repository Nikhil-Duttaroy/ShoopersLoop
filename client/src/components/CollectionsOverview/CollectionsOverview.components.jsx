import { React } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CollectionPreview from "./../CollectionPreview/CollectionPreview.components";
import { SelectCollectionForPreview } from "./../../redux/shop/shop.selector";


import { CollectionsOverviewContainer } from "./CollectionsOverview.styles";

export const CollectionsOverview = ({ collections }) => (
  <CollectionsOverviewContainer>
    {collections.map(({ id, ...otherCollectionProps }) => (
      <CollectionPreview key={id} {...otherCollectionProps} />
    ))}
  </CollectionsOverviewContainer>
);
const mapStateToProps = createStructuredSelector({
  collections: SelectCollectionForPreview,
});

export default connect(mapStateToProps)(CollectionsOverview)