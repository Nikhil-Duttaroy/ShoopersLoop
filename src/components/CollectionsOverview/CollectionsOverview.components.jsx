import { React } from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import CollectionPreview from "./../CollectionPreview/CollectionPreview.components";
import { SelectCollectionForPreview } from "./../../redux/shop/shop.selector";


import './CollectionsOverview.styles.scss'


const CollectionsOverview = ({ collections }) => (
  <div className='collections-overview'>
    {collections.map(({ id, ...othercollectionProps }) => (
      <CollectionPreview key={id} {...othercollectionProps} />
    ))}
  </div>
);

const mapStateToProps = createStructuredSelector({
  collections: SelectCollectionForPreview,
});

export default connect(mapStateToProps)(CollectionsOverview)