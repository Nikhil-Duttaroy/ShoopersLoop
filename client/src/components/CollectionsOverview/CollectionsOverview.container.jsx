// import React from 'react';
import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import {compose} from 'redux' //makes wrapping multiple HOC'S easier

import {
  selectIsCollectionFetching} from "../../redux/shop/shop.selector";
import WithSpinner from "../../components/WithSpinner/WithSpinner.component";
import CollectionsOverview from './CollectionsOverview.components';


const mapStateToProps = createStructuredSelector({
  isLoading: selectIsCollectionFetching,  
});

// const CollectionsOverviewContainer = connect(mapStateToProps)(WithSpinner((CollectionsOverview)))

 const CollectionsOverviewContainer = compose(
  connect(mapStateToProps),
  WithSpinner)
  (CollectionsOverview)


export default CollectionsOverviewContainer;
