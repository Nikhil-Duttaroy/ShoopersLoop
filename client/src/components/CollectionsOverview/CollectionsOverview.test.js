import React from "react";
import { shallow } from "enzyme";
import {CollectionsOverview} from './CollectionsOverview.components';
import { CollectionsOverviewContainer } from './CollectionsOverview.styles';



it("should render CollectionsOverview component", () => {
  expect(shallow(<CollectionsOverview collections={[]} />)).toMatchSnapshot();
});

it("should render CollectionsOverviewContainer component", () => {
  expect(
    shallow(<CollectionsOverviewContainer/>)
  ).toMatchSnapshot();
});
