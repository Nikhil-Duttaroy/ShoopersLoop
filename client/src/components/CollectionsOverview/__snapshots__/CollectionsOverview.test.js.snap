// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CollectionsOverview component 1`] = `<styled.div />`;

exports[`should render CollectionsOverviewContainer component 1`] = `
<div
  className="sc-bqiQRQ dzsFsI"
/>
`;
